frontend/src/main/scala/bloop/engine/Build.scala

```diff
@@ -1,13 +1,19 @@
 package bloop.engine
 
 import bloop.Project
-import bloop.io.AbsolutePath
+import bloop.io.{AbsolutePath, DirChecksum}
 
 final case class Build(origin: AbsolutePath, projects: List[Project]) {
+
+  private val originChecksum = DirChecksum(origin, Project.loadPattern)
```

Comment:
Following up on the previous comment on mutability, I think it would be better to have the checksums live in the state itself and be independent of this data structure.

Reply:
What are your arguments for that? `Build` is just a wrapper around the list of projects, so I think it makes more sense to keep here the information needed to tell whether the `Build` is outdated or not.

Reply:
As I see it, we have `Build` and then we have the state of `Build`. Their representation should be separate so that we can reason about it better, and so that we can update or check its state without necessarily updating the build.